packages/flame/test/components/component_lifecycle_test.dart

```diff
@@ -47,6 +47,47 @@ void main() {
       );
     });
 
+    flameGame.test('component hasMounted completes', (game) async {
+      final component = _MyComponent();
+      await game.add(component);
+      final hasMounted = component.hasMounted;
```

Comment:
I presume it is not possible to await for this future, since it cannot complete without the game loop running its course -- hence this would become an infinite cycle.

Perhaps it would be useful to have at least some test where we await `component.hasMounted`, a test case that would mimic the intended use case for this feature.

Reply:
I double checked and the future does completes, otherwise the test would fail. I changed to expectLater to be sure, but even the original code was working.

Reply:
But can you do `await component.hasMounted;`? I mean, isn't that how this future is supposed to be used?

Reply:
Yeah, you can await it, the `completes` matcher makes sure that it finishes.

I could change my test to make use of the `await` keyword, but would make the test uglier IMO

Reply:
The point is, right now we have no tests to check whether this future is actually awaitable.

Reply:
How come? This test is doing exactly that 🤔